File: vz03/src/renders/facilities/road/entrance/entrance.cpp
/* NIM/Nama  : 13515AAA / AAA AAA   */
/* File      : Entrance.cpp         */

#include <iostream>
#include "entrance.h"
using namespace std;

/***** ENTRANCE *****/
Entrance::Entrance() : Road('i', WHITE)
{}

Entrance::Entrance(const Entrance& E) : Road(E)
{}

Entrance::~Entrance()
{}

Entrance& Entrance::operator=(const Entrance& E)
{
  Facility::operator=(E);
  return *this;
}

char Entrance::render()
{
	return id;
}

Color Entrance::getColor()
{
	return color;
}

Entrance* Entrance::clone() const {
  return new Entrance(*this);
}

File: virtual-zoo-java/src/facility/road/entrance/Entrance.java
/**
 * @author NIM / Nama : 13515087 / Audry Nyonata
 *  Nama file : Entrance.java
 */

package facility.road.entrance;

import facility.road.Road;

public class Entrance extends Road {
  /**
   * Constructor.
   */
  public Entrance() {
    super();
  }
  
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar.
   * @return id bertipe char
   */
  public char render() {
    return 'i';
  }
  
  /**
   * Mengembalikan warna untuk ditampilkan ke layar.
   * @return color warna renderable
   */
  public String renderWithColor() {
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}